### PR TITLE
Feat: Add impelementaion of add new participant endpoint

### LINKED ISSUE #54: Implement Add new participant endpoint

### PR DESCRIPTION
**Pull request by Mojtabarezaei4**
This also includes #59. Room is needed for a participant to be saved in db therefore signing a participant up for a room can be done here. A Participant hits the endpoint through a specific room anyway.

Closes #54 and #59.

### CODE REVIEW

File: Queueomatic/Queueomatic.Test/ParticipantService_AddParticipant.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace Queueomatic.Test;
+
+public class ParticipantService_AddParticipant
+{
+    [Fact]
+    public async Task AddParticipant_ReturnParticipantDto()
+    {
+        // Arrange
+        var uof = A.Fake<IUnitOfWork>();
```
Comment (FabianAronsson) [inline Queueomatic/Queueomatic.Test/ParticipantService_AddParticipant.cs:16]: uof?

File: Queueomatic/Server/Endpoints/Participant/Add/AddNewParticipantEndpoint.cs [hunk 2]
```diff
@@ -1,32 +1,52 @@
...
+    public AddNewParticipantEndpoint(IConfiguration configuration, IParticipantService participantService)
+    {
+        _configuration = configuration;
+        _participantService = participantService;
+    }
+
     public override void Configure()
     {
         Verbs(Http.POST);
         Routes("/rooms/{roomId}/newParticipant");
```
Comment (FabianAronsson) [inline Queueomatic/Server/Endpoints/Participant/Add/AddNewParticipantEndpoint.cs:23]: Maybe clump verbs and routes into a single one?

Comment (Mojtabarezaei4) [reply Queueomatic/Server/Endpoints/Participant/Add/AddNewParticipantEndpoint.cs]: Forgot to shrink it.

**Comment by Mojtabarezaei4** [comment]
The endpoint itself is not tested by XUnit test. I've tested manually.

**Pull request merged**